core/src/main/scala/dagr/core/execsystem/TaskTracker.scala

```diff
@@ -62,12 +62,9 @@ trait TaskTracker extends TaskManagerLike with LazyLogging {
   /** The next id for a task */
   private var nextId: TaskId = 0
 
-  /** The bi-directional map that stores the relationship between a task id and the task.  Needed to find a task
-    * given its id, since a task knows about its id, but an id does not know about its task. */
-  private val idLookup = mutable.Map[TaskId, Task]()
-
-  /** The bi-directional map that stores the relationship between a task and its execution info. */
-  private val taskAndGraphNode = new BiMap[Task, GraphNode]()
```

Comment:
This is the source of the first bug.  If the input `Task` had the same `hashCode` we could get collisions (overwriting) in the `BiMap`